includes/contribution-meter/class-contribution-meter.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * Newspack Contribution Meter.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Contribution_Meter;
+
+use Newspack\Donations;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Handles contribution meter data retrieval and calculations.
+ */
+class Contribution_Meter {
+
+	/**
+	 * REST route for contribution meter.
+	 */
+	const REST_ROUTE = '/contribution-meter';
+
+	/**
+	 * Cache duration in seconds for donation revenue calculations.
+	 */
+	const CACHE_DURATION = 600;
+
+	/**
+	 * Cache duration option name.
+	 */
+	const CACHE_DURATION_OPTION = 'newspack_contribution_meter_cache_duration';
+
+	/**
+	 * Cache key prefix for contribution meter data.
+	 */
+	const CACHE_KEY_PREFIX = 'newspack_contribution_meter_';
+
+	/**
+	 * Initialize hooks and REST API endpoints.
+	 */
+	public static function init() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_rest_routes' ] );
+	}
+
+	/**
+	 * Register REST API routes for the contribution meter.
+	 */
+	public static function register_rest_routes() {
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			self::REST_ROUTE,
+			[
+				'methods'             => 'POST',
+				'callback'            => [ __CLASS__, 'api_get_contribution_data' ],
+				'permission_callback' => function () {
+					return current_user_can( 'edit_posts' );
+				},
+				'args'                => [
+					'startDate' => [
+						'required'          => true,
+						'type'              => 'string',
+						'validate_callback' => [ __CLASS__, 'validate_date' ],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Validate date format and value.
+	 *
+	 * @param string $date Date string to validate.
+	 * @return bool|WP_Error True if valid, WP_Error if invalid.
+	 */
+	public static function validate_date( $date ) {
+		if ( ! preg_match( '/^\d{4}-\d{2}-\d{2}$/', $date ) ) {
+			return new \WP_Error( 'invalid_date', __( 'Invalid date format. Expected YYYY-MM-DD.', 'newspack-plugin' ) );
+		}
+		$date_obj = \DateTime::createFromFormat( 'Y-m-d', $date );
+		if ( ! $date_obj || $date_obj->format( 'Y-m-d' ) !== $date ) {
+			return new \WP_Error( 'invalid_date', __( 'Invalid date. Please provide a valid date.', 'newspack-plugin' ) );
+		}
+		return true;
+	}
+
+	/**
+	 * REST API callback to get contribution data.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response|WP_Error Response object or error.
```

Comment:
Nit: across this file my IDE is yelling at me that `Newspack\Contribution_Meter\WP_...` does not exist.

Can we either add the root namespace in the comments or switch to `use WP_...` at the top? No strong preference on which.